src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -429,10 +429,19 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 		PlanState *planState = &(scanState->customScanState.ss.ps);
 
 		ExecuteMasterEvaluableFunctions(jobQuery, planState);
+		ExecuteTaskListFunctions(taskList, planState);
 
 		if (deferredPruning)
 		{
-			AssignInsertTaskShardId(jobQuery, taskList);
+			AssignInsertJobShardIds(workerJob);
```

Comment:
Thinking I'm going to need to either merge #1474 to `master` first or rebase _this_ on top of _that_ to satisfy these suggestions.